sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -200,6 +425,308 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
+                                                                                                 Integer timeout,
+                                                                                                 String leaseId,
+                                                                                                 String requestId,
+                                                                                                 CpkInfo cpkInfo,
+                                                                                                 Callback<BlockBlobsStageBlockResponse> callback) {
+        String encryptionKey = null;
+        String encryptionKeySha256 = null;
+        EncryptionAlgorithmType encryptionAlgorithm = null;
+        if (cpkInfo != null) {
+            encryptionKey = cpkInfo.getEncryptionKey();
+            encryptionKeySha256 = cpkInfo.getEncryptionKeySha256();
+            encryptionAlgorithm = cpkInfo.getEncryptionAlgorithm();
+        }
+        //
+        final String comp = "block";
+        String transactionalContentMD5Converted = Base64Util.encodeToString(transactionalContentMD5);
+        String transactionalContentCrc64Converted = Base64Util.encodeToString(transactionalContentCrc64);
+        //
+        int contentLength = blockContent.length;
+        RequestBody body = RequestBody.create(MediaType.get("application/octet-stream"), blockContent);
+
+        if (callback != null) {
+            Call<ResponseBody> call = service.stageBlock(containerName,
+                blobName,
+                base64BlockId,
+                contentLength,
+                transactionalContentMD5Converted,
+                transactionalContentCrc64Converted,
+                body,
+                timeout,
+                leaseId,
+                XMS_VERSION,
+                requestId,
+                comp,
+                encryptionKey,
+                encryptionKeySha256,
+                encryptionAlgorithm);
+
+            executeCall(call, new retrofit2.Callback<ResponseBody>() {
+                @Override
+                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                    if (response.isSuccessful()) {
+                        if (response.code() == 201) {
+                            BlockBlobStageBlockHeaders typedHeader = deserializeHeaders(response.headers(),
+                                BlockBlobStageBlockHeaders.class);
+                            callback.onResponse(new BlockBlobsStageBlockResponse(response.raw().request(),
+                                response.code(),
+                                response.headers(),
+                                null,
+                                typedHeader));
+                        } else {
+                            String strContent = readAsString(response.body());
+                            callback.onFailure(new BlobStorageException(strContent, response.raw()));
```

Comment:
Is this a failure if `response.isSuccessful()` is true?

Reply:
(And apply this question for other instances of this pattern)

Reply:
That is correct.

While okhttp (or generally in http) treat 2xx code as successful, from storage service perspective only successful response status code is 201 for the stage API. This is described in swagger response section for the sample [stage-api](https://github.com/Azure/azure-rest-api-specs/blob/52f4a2bf17f0bbd5649473faea88bd3205e74c24/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-02-02/blob.json#L6032). 

Similarly for other apis as well.